src/parameter.cpp

```diff
@@ -21,65 +21,63 @@
 using namespace std;
 
 #define ERRORLOG cerr << "ERROR in Parameter: "
-#define TRACELOG if (_verbose) cout << "TRACE in Parameter: "
+#define TRACELOG \
+  if (_verbose)  \
+  cout << "TRACE in Parameter: "
 
-namespace acap_runtime {
-
-// Initialize Parameter Service
-bool Parameter::Init(const bool verbose) {
-  _verbose = verbose;
-  TRACELOG << "Init" << endl;
-  return true;
-}
-
-// Logic and data behind the server's behavior.
-Status Parameter::GetValues(ServerContext* context,
-    ServerReaderWriter<Response, Request>* stream) {
+namespace acap_runtime
+{
+  Parameter::~Parameter()
+  {
+    if (ax_parameter != NULL)
+      ax_parameter_free(ax_parameter);
+  }
 
-    char parhand_result[BUFSIZ];
+  // Initialize Parameter Service
+  bool Parameter::Init(const bool verbose)
```

Comment:
If Parameter::Init was instead the class constructor, throwing runtime_error on failure, there wouldn't be a need to check ax_parameter == NULL before assignment and it wouldn't have to be checked at destruction. But maybe it's not worth the effort.